engine/src/scheme.rs

```diff
@@ -51,8 +75,29 @@ impl<'i, 's> LexWith<'i, &'s Scheme> for Field<'s> {
 
         let name = span(initial_input, input);
 
+        let mut path = vec![];
+
+        while let Ok(rest) = expect(input, "[") {
+            input = skip_space(rest);
+
+            let key = RhsValue::lex_with(input, Type::Bytes)?;
```

Comment:
Better to use destructuring here.

Reply:
I will change to `let (key, mut input) = RhsValue::lex_with`

Reply:
Hmm, nope, that rebinds input inside the loop which isn't what we want.

Reply:
Can we de-structure without creating a new binding for input?

Reply:
Unfortunately, no, but it's okay to have temporary variable (I usually use `rest` in the codebase as an alternative to `input` when talking about "rest of input").

Still improves readability over `.0` / `.1`.